### DISCUSSION TITLE
Creation and Identification of admin roles

### DISCUSSION THREAD

**Discussion by githubpiyush** [Sunbird-coKreat/Community#33] [General Discussions]
@krgauraw

1. At present what all admin roles are used in creation portal ?
2. How can one create various admin roles in creation portal?
3. What is the difference between sourceOrgAdmin, sourceAdmin and orgAdmin?
4. how can we identify from user profile which user has what role?

cc - @kiranharidas187 @vijiurs @aks30 @pradeepkumarcm96

**Comment by Nadiya496** [comment]
@githubpiyush 

1. Currently in the creation portal we have sourcing org admin and Contribution org admin.
2. Consumption portal , user with "ORG_ADMIN" role becomes the sourcing org admin in the creation portal. By default user also gets the contribution org admin role on the creation portal when the project is created, published for the 1st time. Apart from that external user (Non consumption portal user ) can also become part of cokreat portal by enrolling himself as a contributor. If he selects organisation in the pop up, the contirbution org thus gets created.
![image](https://user-images.githubusercontent.com/53341543/201258883-2ec5bff7-ea20-4954-b9e3-ad5dd4e6e247.png)

3. Org admin in the consumption repo has the access to become the sourcing org admin in the creation portal.
4. Sourcing org admin will have the "ORG_ADMIN" role in the consumption repo. Contribution org admin will have user with "admin" role in the open saber registry. 
cc - @kiranharidas187 @vijiurs @aks30 @pradeepkumarcm96 @pallakartheekreddy @surendrasinghs

**Reply by githubpiyush** [reply]
@Nadiya496 
Let me know if this understanding is correct and we are on the same page. 


> 2. Consumption portal , user with "ORG_ADMIN" role becomes the sourcing org admin in the creation portal. By default user also gets the contribution org admin role on the creation portal when the project is created, published for the 1st time. Apart from that external user (Non consumption portal user ) can also become part of cokreat portal by enrolling himself as a contributor. If he selects organisation in the pop up, the contirbution org thus gets created.

The procedure of creation of admin roles are taken care by subird lern team. 
Any external user can come to contribute side of creation portal and create a organisation then by default he becomes the contributor org admin, in this case we don't have sourcing org admin role for this organisation. 

> 3. Org admin in the consumption repo has the access to become the sourcing org admin in the creation portal.
There is only sourcing org admin in creation portal.


> 4. Sourcing org admin will have the "ORG_ADMIN" role in the consumption repo. Contribution org admin will have user with "admin" role in the open saber registry.

If a user has a role as ORG_ADMIN then he is a sourcingOrgAdmin.
When a sourcingOrgAdmin creates a project in VDN sourcing portal and publishes it then by default he becomes contributorOrgAdmin.

A contributorOrgAdmin can make a user (who has a role content creator/reviewer in consumption) contributorOrgAdmin by changing his role from user to admin in the manage user tab.

![image](https://user-images.githubusercontent.com/35293206/201344561-2afd62c6-2699-4da7-85ec-25685deb4f94.png)


A contributorOrgAdmin can make a external user contributorOrgAdmin by changing his role from user to admin in the manage user tab.
Once we make the external user a contributorOrgAdmin then his role will be updated as admin in the open saber registry(This contains all the external user's information in creation portal).

**Reply by Nadiya496** [reply]
Yes @githubpiyush
